File: src/resources/locales.ts
import { OllangClient } from '../client';
import { Language, LanguageValidationResult } from '../types';

/**
 * The platform language catalogue.
 *
 * Order and project creation match language codes exactly and reject anything
 * not in the catalogue, so resolve uncertain codes here rather than guessing.
 * Codes are mostly ISO 639-1 with regional and platform-specific variants
 * (`pt` is Portuguese (Brazil), `pt-PT` is Portugal).
 */
export class Locales {
  constructor(private client: OllangClient) {}

  /** Lists supported languages with their regional variants. */
  async languages(): Promise<Language[]> {
    return this.client.get<Language[]>('/integration/locales/languages');
  }

  /** Searches languages by name, native name or code. */
  async search(query: string): Promise<Language[]> {
    return this.client.get<Language[]>('/integration/locales/search', { q: query });
  }

  /**
   * Checks a language code against the catalogue.
   *
   * Reports whether the code is accepted, its parsed language and region, and
   * why it failed if it did.
   */
  async validate(tag: string): Promise<LanguageValidationResult> {
    return this.client.get<LanguageValidationResult>(
      `/integration/locales/validate/${encodeURIComponent(tag)}`
    );
  }
}

File: java/src/main/java/com/ollang/sdk/resources/Locales.java
package com.ollang.sdk.resources;

import com.google.gson.JsonElement;
import com.ollang.sdk.OllangClient;
import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;
import java.util.LinkedHashMap;
import java.util.Map;

/**
 * The platform language catalogue.
 *
 * <p>Order and project creation match language codes exactly and reject anything not in the
 * catalogue, so resolve uncertain codes here rather than guessing. Codes are mostly ISO 639-1
 * with regional and platform-specific variants ({@code pt} is Portuguese (Brazil), {@code pt-PT}
 * is Portugal).
 */
public class Locales {

  private final OllangClient client;

  public Locales(OllangClient client) {
    this.client = client;
  }

  /** Lists supported languages with their regional variants. */
  public JsonElement languages() {
    return client.get("/integration/locales/languages");
  }

  /** Searches languages by name, native name or code. */
  public JsonElement search(String query) {
    Map<String, String> params = new LinkedHashMap<>();
    params.put("q", query);
    return client.get("/integration/locales/search", params);
  }

  /**
   * Checks a language code against the catalogue.
   *
   * @return whether the code is accepted, its parsed language and region, and why it failed if it
   *     did
   */
  public JsonElement validate(String tag) {
    return client.get(
        "/integration/locales/validate/" + URLEncoder.encode(tag, StandardCharsets.UTF_8));
  }
}
